{
  try {
    _countDownLatch.await();
    _executorService.execute(new MessageCallBackJob(message));
  }
 catch (  InterruptedException ie) {
    _log.error("Latch opened prematurely by interruption. Dependence may " + "not be ready.");
  }
catch (  RejectedExecutionException ree) {
    _log.error("Unable to handle received message " + message,ree);
  }
}
